src/lua/protobuf.lua

```diff
@@ -0,0 +1,316 @@
+local ffi = require('ffi')
+local encoder = require('internal.protobuf.wireformat')
+local protocol_mt
+
+local min_field_id = 1
+local reserved_field_id_min = 19000
+local reserved_field_id_max = 19999
+local max_field_id = 2^29 - 1
+local max_int64 = 2^64
+local max_int32 = 2^32
```

Comment:
Let's uppercase the constants.

BTW, if you can link a source of these values, it would be nice to add the links into a comment. At least, `max_field_id` and `reserved_*` looks like they're originating from some document. 1, 2^32, 2^64 are quite self-explanatory, they, of course, don't need a comment.

BTW 2, 2^32 is max **uint32**, not max int32. The constant name may be confusing. The same for `max_int64`.

I'm also worring about a constant for 2^64 of a `number` type, it may be imprecise. However, I don't dive deep enough to say, whether its usage actually may lead to a problem.

Reply:
Changed to uppercase, added a comment with a link to the protobuf documentation because our API does not contain information about reserved ids (Maybe needed to add?)

Reply:
> our API does not contain information about reserved ids (Maybe needed to add?)

It is OK for me to just perform the checks based on the protocol specification. No need to export the constants, because it is very unlikely that someone generates the IDs automatically.